internal/core/thirdparty/tantivy/tantivy-binding/src/index.rs

```diff
@@ -0,0 +1,329 @@
+            index_writer: Some(index_writer),
+            reader: None,
+        };
+        ret.create_reader();
+        ret
+    }
+
+    pub fn load(path: &str) -> IndexWrapper {
+        let dir = MmapDirectory::open(path).unwrap();
+        let index = Index::open(dir).unwrap();
+        let field = index.schema().fields().next().unwrap().0;
+        let schema = index.schema();
+        let field_name = schema.get_field_name(field);
+        let mut ret = IndexWrapper {
+            field_name: Some(String::from_str(field_name).unwrap()),
+            field: Some(field),
+            data_type: None,
+            path: None,
+            index: Some(index),
+            index_writer: None,
+            reader: None,
+        };
+        ret.create_reader();
+        ret.reload();
+        ret
+    }
+
+    pub fn add_i8(&mut self, data: i8) {
+        self.add_i64(data.into())
+    }
+
+    pub fn add_i16(&mut self, data: i16) {
+        self.add_i64(data.into())
+    }
+
+    pub fn add_i32(&mut self, data: i32) {
+        self.add_i64(data.into())
+    }
+
+    pub fn add_i64(&mut self, data: i64) {
+        // println!("{} was input", data);
+        self.index_writer
+            .as_mut()
+            .unwrap()
+            .add_document(doc!(self.field.unwrap() => data))
+            .unwrap();
+    }
+
+    pub fn add_f32(&mut self, data: f32) {
+        self.add_f64(data.into())
+    }
+
+    pub fn add_f64(&mut self, data: f64) {
+        // println!("{} was input", data);
+        self.index_writer
+            .as_mut()
+            .unwrap()
+            .add_document(doc!(self.field.unwrap() => data))
+            .unwrap();
+    }
+
+    pub fn add_bool(&mut self, data: bool) {
+        // println!("{} was input", data);
+        self.index_writer
+            .as_mut()
+            .unwrap()
+            .add_document(doc!(self.field.unwrap() => data))
+            .unwrap();
+    }
+
+    pub fn add_keyword(&mut self, data: &str) {
+        // println!("{} was input", data);
+        self.index_writer
+            .as_mut()
+            .unwrap()
+            .add_document(doc!(self.field.unwrap() => data))
+            .unwrap();
+    }
+
+    pub fn finish(&mut self) {
+        self.index_writer.as_mut().unwrap().commit().unwrap();
+        block_on(self.index_writer.as_mut().unwrap().garbage_collect_files()).unwrap();
+        self.reload();
+    }
+
+    pub fn count(&mut self) -> u32 {
+        let metas = self
+            .index
+            .as_mut()
+            .unwrap()
+            .searchable_segment_metas()
+            .unwrap();
+        let mut sum: u32 = 0;
+        for meta in metas {
+            sum += meta.max_doc();
+        }
+        sum
+    }
+
+    fn search(&mut self, q: &dyn Query) -> Vec<u32> {
+        let searcher = self.reader.as_mut().unwrap().searcher();
+        let cnt = self.count();
+        let hits = searcher
+            .search(q, &TopDocs::with_limit(cnt as usize))
```

Comment:
Does search always returns Ok() at runtime?
no Err() handling but unwrap will cause the program crash at runtime.